### DISCUSSION TITLE
SSR and Firebase updates in real time

### DISCUSSION THREAD

**Discussion by jonsnowpt** [vercel/next.js#18776] [Help] [answered]
Is it possible using SSR to have data updated in real-time using Firebase? Or is the page simply static with no possibility of receiving new data updated in real-time on the screen?

**Comment by jonsnowpt** [comment]
No one tried this?

**Comment by lfades** [comment, marked as answer]
You can use CSR (Client side-rendering) to do this, it can be done after hydration (usually inside `useEffect`). If you also wish to update the content of a statically generated page, [Incremental Static Revalidation](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration) is the way to go there, you can set it to be as low as 1 second.

**Reply by jonsnowpt** [reply]
Thank you.

So using Incremental Static Regeneration allows if the user does not refresh the page, data is updated on the screen? For example, I have a table of live football scores, which has constantly updated data, with this option the user can see the data update on the screen without having to refresh the page every second?

Of course, keeping the server-side rendering for SEO purposes.
